components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt

```diff
@@ -86,8 +87,33 @@ abstract class AbstractFetchDownloadService : Service() {
         @GuardedBy("context") var status: DownloadJobStatus,
         var foregroundServiceId: Int = 0,
         var downloadDeleted: Boolean = false,
-        var notifiedStopped: Boolean = false
-    )
+        var notifiedStopped: Boolean = false,
+        @Volatile var notificationStatus: DownloadJobStatus? = null,
+        var lastNotificationUpdate: Long = 0L
+    ) {
+        internal fun canUpdateNotification(status: DownloadJobStatus): Boolean {
+            return isUnderNotificationUpdateLimit() && (!isNotificationInSync(status) || !notifiedStopped)
```

Comment:
Hm it seems this should be `(!isNotificationInSync(status) && !notifiedStopped). If the notification is in sync and notification are not stopped you still don't want to notify, right?